### DISCUSSION TITLE
Is Joint VIMP dependent on the number of variables in each group?

### DISCUSSION THREAD

**Discussion by NelM3004** [kogalur/randomForestSRC#359] [Q&A]
Hey, 

I want to compare the VIMP of different variables as well as the importance of different groups of these variables. However, the number of variables per group varies between two and seven. So my question is: Are the results for the joint VIMP (strongly) influenced by the number of variables in each group? And if yes, is there a good approach to correct this or to calculate "adjusted VIMPs"? 

In most of my random forest models there is a clear trend in the VIMP, so that only a few variables show a high variable importance and thus only these variables would contribute to the joint VIMP in each group (to a higher degree). Simply dividing the VIMP by the number of variable per group would in these cases be the wrong approach, wouldn't it?

Thanks for any information, advice or thoughts on this question!

Best regards
Nele

**Comment by ishwaran** [comment]
Even though it might seem like it, permutation VIMP does not have the theoretical property of isolating the effect of a specific variable (or variables in the case of joint VIMP).  Rather it gives an informal assessment of the predictive power of a variable that can be very helpful for ranking variables and identifying variables of importance.

If you really want to isolate the effect of a variable, or group of variables, you should use [holdout.vimp](https://www.randomforestsrc.org/reference/holdout.vimp.rfsrc.html) which holds out the variable in the tree construction and therefore is a more rigorous way to assess the direct effect size.  Note that this method will tend to be much more conservative than permutation VIMP.

**Reply by NelM3004** [reply]
Thank you very much for the quick response! Actually I am interested in comparing the hierarchy and the ranking of these groups of variables and not necessarily interested in the direct effect size. However, I was wondering if the joint VIMP (and it would be the same for joint holdout.vimp) is (strongly) influenced by the number of variables per group?
